mycroft/audio/service.py

```diff
@@ -0,0 +1,224 @@
+        self.status.bind(self.bus)
+
+        try:
+            self.init_messagebus()
+            self.audio = AudioService(self.bus)
+        except Exception as e:
+            self.status.set_error(e)
+
+    def run(self):
+        if self.audio.wait_for_load():
+            if len(self.audio.service) == 0:
+                LOG.warning('No audio backends loaded! '
+                            'Audio playback is not available')
+                LOG.info("Running audio service in TTS only mode")
+        # If at least TTS exists, report ready
+        if self.tts:
+            self.status.set_ready()
+        else:
+            self.status.set_error('No TTS loaded')
+
+    def handle_speak(self, event):
+        """Handle "speak" message
+
+        Parse sentences and invoke text to speech service.
+        """
+
+        # if the message is targeted and audio is not the target don't
+        # don't synthesise speech
+        event.context = event.context or {}
+        if event.context.get('destination') and not \
+                ('debug_cli' in event.context['destination'] or
+                 'audio' in event.context['destination']):
+            return
+
+        # Get conversation ID
+        if event.context and 'ident' in event.context:
+            ident = event.context['ident']
+        else:
+            ident = 'unknown'
+
+        with self.lock:
+            stopwatch = Stopwatch()
+            stopwatch.start()
+
+            utterance = event.data['utterance']
+            listen = event.data.get('expect_response', False)
+            self.mute_and_speak(utterance, ident, listen)
+
+            stopwatch.stop()
+
+        report_timing(ident, 'speech', stopwatch,
+                      {'utterance': utterance,
+                       'tts': self.tts.__class__.__name__})
+
+    def _maybe_reload_tts(self):
+        config = Configuration().get("tts", {})
+
+        # update TTS object if configuration has changed
+        if not self._tts_hash or self._tts_hash != config.get("module", ""):
+            if self.tts:
+                self.tts.shutdown()
+            # Create new tts instance
+            LOG.info("(re)loading TTS engine")
+            self.tts = TTSFactory.create(config)
+            self.tts.init(self.bus)
+            self._tts_hash = config.get("module", "")
+
+        # if fallback TTS is the same as main TTS dont load it
+        if config.get("module", "") == config.get("fallback_module", ""):
+            return
+
+        if not self._fallback_tts_hash or \
+                self._fallback_tts_hash != config.get("fallback_module", ""):
+            if self.fallback_tts:
+                self.fallback_tts.shutdown()
+            # Create new tts instance
+            LOG.info("(re)loading fallback TTS engine")
+            self._get_tts_fallback()
+            self._fallback_tts_hash = config.get("fallback_module", "")
+
+    def mute_and_speak(self, utterance, ident, listen=False):
```

Comment:
do we want to keep this name since we are not bound by compat anymore?

where is the muting being done? maybe just call it execute to match the tts method?

Reply:
Yeah, I guess it's just calling `execute`, which used to mute the mic and speak, but that's configurable now